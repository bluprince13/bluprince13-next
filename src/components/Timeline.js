import Timeline from '@mui/lab/Timeline'
import TimelineItem from '@mui/lab/TimelineItem'
import TimelineSeparator from '@mui/lab/TimelineSeparator'
import TimelineConnector from '@mui/lab/TimelineConnector'
import TimelineContent from '@mui/lab/TimelineContent'
import TimelineOppositeContent from '@mui/lab/TimelineOppositeContent'
import TimelineDot from '@mui/lab/TimelineDot'
import Typography from '@mui/material/Typography'
import Icon from '@mui/material/Icon'

const Item = ({ marker, title, description, iconName, color }) => (
    <TimelineItem>
        <TimelineOppositeContent
            sx={{ m: 'auto 0' }}
            align="right"
            variant="body2"
            color="text.secondary"
        >
            {marker}
        </TimelineOppositeContent>
        <TimelineSeparator>
            <TimelineConnector />
            <TimelineDot color={color ? color : 'primary'}>
                <Icon>{iconName}</Icon>
            </TimelineDot>
            <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent sx={{ py: '12px', px: 2, m: 'auto 0' }}>
            <Typography variant="h6" component="span">
                {title}
            </Typography>
            <Typography>{description}</Typography>
        </TimelineContent>
    </TimelineItem>
)
export default function CustomizedTimeline({ items }) {
    return (
        <Timeline position="alternate">
            {items.map(({ marker, title, description, iconName, color }) => (
                <Item
                    key={marker}
                    marker={marker}
                    title={title}
                    description={description}
                    iconName={iconName}
                    color={color}
                />
            ))}
        </Timeline>
    )
}
