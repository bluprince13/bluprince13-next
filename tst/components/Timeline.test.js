import Timeline from '@Components/Timeline'
import { render } from '@testing-library/react'

describe('Timeline', () => {
    const setup = () => {
        const queries = render(
            <Timeline
                items={[
                    {
                        marker: 'marker1',
                        title: 'With description',
                        description: 'description1',
                        iconName: 'start',
                        color: 'info'
                    },
                    {
                        marker: 'marker2',
                        title: 'Without description',
                        iconName: 'end',
                        color: 'error'
                    }
                ]}
            />
        )
        return queries
    }

    it('renders correctly', () => {
        const { asFragment } = setup()

        expect(asFragment()).toMatchSnapshot()
    })
})
