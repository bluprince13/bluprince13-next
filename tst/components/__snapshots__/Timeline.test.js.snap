// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="MuiTimeline-root MuiTimeline-positionAlternate css-55vwbm-MuiTimeline-root"
  >
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionAlternate css-1sohoek-MuiTimelineItem-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionAlternate css-l4x062-MuiTypography-root-MuiTimelineOppositeContent-root"
      >
        marker1
      </div>
      <div
        class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      >
        <span
          class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
        />
        <span
          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledInfo css-1mqyah4-MuiTimelineDot-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            start
          </span>
        </span>
        <span
          class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionAlternate css-obnzm6-MuiTypography-root-MuiTimelineContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          With description
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          description1
        </p>
      </div>
    </li>
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionAlternate css-1sohoek-MuiTimelineItem-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionAlternate css-l4x062-MuiTypography-root-MuiTimelineOppositeContent-root"
      >
        marker2
      </div>
      <div
        class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      >
        <span
          class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
        />
        <span
          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledError css-ia85ya-MuiTimelineDot-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            end
          </span>
        </span>
        <span
          class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionAlternate css-obnzm6-MuiTypography-root-MuiTimelineContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Without description
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
